include/aos/common/tools/fs.hpp

```diff
@@ -364,33 +347,31 @@ class FS {
             return Error(errno);
         }
 
-        auto OnError = [fd]() {
-            auto err = errno;
-
+        auto OnError = [fd](const aos::Error& err) {
```

Comment:
We have `DeferRelease` helper to release C resource. Use it here.

Reply:
fixed